src/os/ui/alert/alertpopup.js

```diff
@@ -127,35 +101,33 @@ export class Controller {
    * @private
    */
   handleAlertEvent_(event) {
-    // deduping for floods of identical messages
-    if (this.lastEvent_) {
-      if (event.getMessage() == this.lastEvent_.getMessage()) {
-        if (!this.dupeDelay_) {
-          this.dupeDelay_ = new Delay(this.resetDupes_, Controller.ALERT_TIMER, this);
-        }
-        this.dupeDelay_.start();
-        this.dupeAlerts_++;
-
-        var limit = event.getLimit();
-        if (this.dupeAlerts_ >= limit && !this.dupeOverflow_) {
-          this.dupeOverflow_ = true;
-
-          // only display the duplicate alert message if the default limit was reached
-          if (limit == DEFAULT_LIMIT) {
-            var dupeEvent = new AlertEvent(
-                'Too many duplicate alerts received! Open the alert window for details.',
-                AlertEventSeverity.WARNING);
-            this.displayAlert_(dupeEvent);
-          }
-        }
-      } else {
-        this.resetDupes_();
+    var display = true;
+    var newMessage = event.getMessage();
+
+    // If the popup is a duplicate, consolidate it
+    if (newMessage.length > 0 && newMessage == this.lastMessage_) {
+      // See if a popup for the current alert is showing
+      var alertPopup = null;
+      if (newMessage) {
+        alertPopup = this['alertPopups'].find((popupObj) => popupObj['msg'] == newMessage);
+      }
+
+      if (alertPopup) {
+        // Update the count of the popup
+        alertPopup['count'] += event.getCount();
```

Comment:
Sometimes this count update isn't reflected in the UI because the Angular scope isn't applied if `displayAlert_` isn't called. We should call `apply(this.scope_);` at the end of this block to make sure the new count is shown in the UI.

Reply:
This was a quick change, so I added a commit for it. All good now!